module-bsp/board/rt1051/bsp/eink/EinkDisplay.cpp

```diff
@@ -62,9 +98,66 @@ namespace service::eink
         delete[] currentWaveform.LUTDData;
     }
 
-    EinkStatus_e EinkDisplay::resetAndInit()
+    EinkStatus EinkDisplay::updateDisplay(std::uint8_t *frameBuffer, EinkRefreshMode refreshMode)
+    {
+        return update(frameBuffer);
+    }
+
+    EinkStatus EinkDisplay::refreshDisplay(EinkRefreshMode refreshMode)
+    {
+        const auto isDeepRefresh = refreshMode == EinkRefreshMode::REFRESH_DEEP;
+        return refresh(isDeepRefresh ? EinkDisplayTimingsDeepCleanMode : EinkDisplayTimingsFastRefreshMode);
+    }
+
+    EinkStatus EinkDisplay::prepareDisplay(EinkRefreshMode refreshMode, WaveformTemperature behaviour)
+    {
+        EinkStatus status;
+
+        powerOn();
+
+        const auto temperature =
+            behaviour == WaveformTemperature::KEEP_CURRENT ? getLastTemperature() : EinkGetTemperatureInternal();
+
+        if (refreshMode == EinkRefreshMode::REFRESH_DEEP) {
+            status = setWaveform(EinkWaveforms_e::EinkWaveformGC16, temperature);
+            if (status == EinkStatus::EinkOK) {
```

Comment:
`status` could have its scope limited to this place and then the whole function could benefit from earlier `return`s and less braces.